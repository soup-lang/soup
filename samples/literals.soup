x: 1000000000000 * 2 % 31
console.log(x)
console.log(1000000000000 * 2 % 37)
y: 8.0 * 1000000000000 / 2500000000000
console.log(y)

console.log((1 << 30) + (1 << 30))
console.log(((1 << 30) + (1 << 30)) << 1)
console.log(-1 >> 1)
console.log(-1 >>> 1)

z: 1 + 2
console.log(z.toString())

console.log('Binary ' + 'logic')
console.log(false || false)
console.log(false || true)
console.log(true || false)
console.log(true || true)
console.log(false && false)
console.log(false && true)
console.log(true && false)
console.log(true && true)
