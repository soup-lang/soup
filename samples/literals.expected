8
2
3.2
2147483648
0
-1
2147483647
3
Binary logic
false
true
true
true
false
false
false
true
